File: aspnet-core/src/TicketTracker.Application/Tickets/Dto/TicketDto.cs
using Abp.Application.Services.Dto;
using Abp.AutoMapper;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TicketTracker.Activities.Dto;
using TicketTracker.Components.Dto;
using TicketTracker.Entities;
using TicketTracker.Statuses.Dto;
using TicketTracker.Works.Dto;

namespace TicketTracker.Tickets.Dto {
    [AutoMap(typeof(Ticket))]
    public class TicketDto : AuditedEntityDto<int> {
        [Required]
        public string Title { get; set; }
        public string Description { get; set; }
        public TicketPriority Priority { get; set; }
        public TicketType Type { get; set; } 
         
        public SimpleStatusDto Status { get; set; }
        public SimpleActivityDto Activity { get; set; }
        public List<SimpleWorkDto> Works { get; set; }
    }
}

File: reactjs/src/services/ticket/dto/ticketDto.ts
import { AuditedEntityDto } from "../../dto/auditedEntityDto";
import {TicketPriority} from './ticketPriority';
import {TicketType} from './ticketType';
import {SimpleStatusDto} from '../../status/dto/simpleStatusDto';
import {SimpleActivityDto} from '../../activity/dto/simpleActivityDto';
import {SimpleWorkDto} from '../../work/dto/simpleWorkDto';
import { SimpleComponentDto } from "../../component/dto/simpleComponentDto";
import { SimpleProjectDto } from "../../project/dto/simpleProjectDto";

export class TicketDto extends AuditedEntityDto { 
    title!: string; 
    description?: string; 
    priority!: TicketPriority;
    type!: TicketType;

    component!: SimpleComponentDto;
    project!: SimpleProjectDto;

    status?: SimpleStatusDto;
    activity?: SimpleActivityDto;
    works?: SimpleWorkDto[];
}